examples/call/contracts/Connected.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.26;
+
+import {RevertContext} from "@zetachain/protocol-contracts/contracts/Revert.sol";
+import "@zetachain/protocol-contracts/contracts/evm/GatewayEVM.sol";
+
+contract Connected {
+    GatewayEVM public immutable gateway;
+
+    event RevertEvent(string, RevertContext);
+    event HelloEvent(string, string);
+
+    error TransferFailed();
+    error ApprovalFailed();
+
+    modifier onlyGateway() {
+        require(msg.sender == address(gateway), "Caller is not the gateway");
```

Comment:
transfer and approval are with custom errors and this with require. move all to the same (I prefer custom errors)

Reply:
c89c6da